src/js/popup/popup.js

```diff
@@ -941,82 +941,87 @@
+            otherCheckbox: document.querySelector('input[name="issue-case-other"]'),
+          }
+          
+          // Set the form as invalid
+          formData.form.dataset.formIsValid = false;
+          
+          popup.panel.webcompat.setURLwithIssue(formData);
+          popup.panel.webcompat.setCheckboxListeners(formData);
+          popup.panel.webcompat.showReportInputOtherTextField(formData);
 
-          checkboxes.forEach(checkbox => {
-            // Override arg must be present to override
-            if (toggleOverride === null) {
-              checkbox.required = !checkbox.required;
-            } else {
-              checkbox.required =  toggleOverride;
-            }
+          const reportForm = document.querySelector(".report-issue-content");
+          reportForm.addEventListener("submit", async (event) => {
+            await popup.panel.webcompat.handleReportIssueFormSubmission(event, formData);
           });
+
+          const reportContinueButton = document.querySelector(".report-continue");
+          reportContinueButton.addEventListener("click", popup.events.backClick, false);
         },
-        setRequiredCheckboxListeners: () => {
-          const reportIssueSubmitBtn = document.querySelector(".report-issue-submit-btn");
-          const inputFieldUrl = document.querySelector('input[name="issue_on_domain"]');
-          const checkboxes = document.querySelectorAll('.report-section ul li input');
-          const otherTextField = document.querySelector('input[name="other_issue"]');
-          const isChecked = (element) => element.checked;
+        setCheckboxListeners: (formData) => {
+          const checkboxes = formData.checkboxes;
           
           checkboxes.forEach(checkbox => {
-            checkbox.required = true;
             checkbox.addEventListener("change", ()=> {              
-              // If the user has selected at least one checkbox, and has filled out the website URL input, make the form submittable
-              if ([...checkboxes].some(isChecked) && inputFieldUrl.value && popup.utilities.isSortaAURL(inputFieldUrl.value) && checkbox.name !== "issue-case-other") {
-                reportIssueSubmitBtn.disabled = false;
-                popup.panel.webcompat.toggleRequiredCheckboxListeners(false);
-              } else {
-                reportIssueSubmitBtn.disabled = true;
-                popup.panel.webcompat.toggleRequiredCheckboxListeners(true);
-              }
-
-              // Custom logic if the user clicks the "other" box
-              if (checkbox.name === "issue-case-other" && checkbox.checked && otherTextField.value && popup.utilities.isSortaAURL(otherTextField.value)) {
-                reportIssueSubmitBtn.disabled = false;
-                popup.panel.webcompat.toggleRequiredCheckboxListeners(false);
-              }
+              popup.panel.webcompat.validateForm(formData);
```

Comment:
You'll see this function `popup.panel.webcompat.validateForm()` called every where there's a user input event that could result in a valid form (type into an input, toggle checkbox).